contracts/token/ERC721/ERC721.sol

```diff
@@ -180,16 +168,15 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
      * @return uint256 token ID at the given index of the tokens list owned by the requested address
      */
     function tokenOfOwnerByIndex(address owner, uint256 index) public view override returns (uint256) {
-        require(index < balanceOf(owner), "ERC721Enumerable: owner index out of bounds");
-        return _ownedTokens[owner][index];
+        return _holderTokens[owner].at(index);
     }
 
     /**
      * @dev Gets the total amount of tokens stored by the contract.
      * @return uint256 representing the total amount of tokens
      */
     function totalSupply() public view override returns (uint256) {
-        return _allTokens.length;
+        return _tokenOwners.length();
```

Comment:
This makes me thing that `lenght` might not be the best name. This returns the number of keys, which in this case are tokenIds. Though perhaps what is wrong is the name of the `_tokenOwners` state variable.

Reply:
I don't know how to fix this but we can at least add a comment clearing up the potential confusion.